app/questionnaire/placeholder_renderer.py

```diff
@@ -57,29 +55,36 @@ def render_placeholder(self, placeholder_data, list_item_id):
             list_store=self._list_store,
         )
 
-        if "text_plural" in placeholder_data:
-            plural_schema = placeholder_data["text_plural"]
+        mutable_placeholder_data = QuestionnaireSchema.get_mutable_deepcopy(
+            placeholder_data
+        )
+
+        if "text_plural" in mutable_placeholder_data:
+            plural_schema = mutable_placeholder_data["text_plural"]
             count = self.get_plural_count(plural_schema["count"])
 
             plural_form_key = get_plural_form_key(count, self._language)
-            placeholder_data["text"] = plural_schema["forms"][plural_form_key]
+            mutable_placeholder_data["text"] = plural_schema["forms"][plural_form_key]
 
-        if "text" not in placeholder_data and "placeholders" not in placeholder_data:
+        if (
+            "text" not in mutable_placeholder_data
+            and "placeholders" not in mutable_placeholder_data
+        ):
             raise ValueError("No placeholder found to render")
 
         transformed_values = placeholder_parser(placeholder_data["placeholders"])
 
-        return placeholder_data["text"].format(**transformed_values)
+        return mutable_placeholder_data["text"].format(**transformed_values)
 
     def render(self, dict_to_render, list_item_id):
         """
         Transform the current schema json to a fully rendered dictionary
         """
-        rendered_data = deepcopy(dict_to_render)
-        pointers = find_pointers_containing(rendered_data, "placeholders")
+        dict_to_render = QuestionnaireSchema.get_mutable_deepcopy(dict_to_render)
```

Comment:
`__deepcopy__` should be implemented on QuestionnaireSchema:
```suggestion
        dict_to_render = deepcopy(QuestionnaireSchema(dict_to_render))
```